Libraries/Components/ScrollView/ScrollView.js

```diff
@@ -1006,10 +1013,6 @@ const ScrollView = createReactClass({
   },
 });
 
-const TypedScrollView = ((ScrollView: any): Class<
```

Comment:
The reason we were exporting this is because ScrollView still uses createReactClass and is thus untyped. I believe with the change in this PR the Props type is now unused, I think this was the only callsite.

Reply:
Oh man, how could I miss that :o. I could migrate this to class component by duplicating ScrollResponder.Mixin (it's only used in deprecated ListView, so the duplication is going to be removed soon anyway)

Reply:
Migrating it to an ES6 class and not using mixins would be great and something we need to do anyways. It probably makes sense to do that in a separate PR as we've seen those be pretty tricky. cc @empyrical as he has converted a few other ones and ran into some edge cases. If you two work together to work out any problems you come across I'd be super excited!

Do you think that PR needs to happen first?

Reply:
Yea, I agree that it would be better to land mixins removal for this component first and rebase this one on top of it.
I'll look up @empyrical's work prior to that, to see what issues I can expect :D

Reply:
The easiest thing to do is probably to do something like:

```
this._scrollResponder = new ScrollResponder();
```

And then in `ScrollView`'s `componentWillMount` call `this._scrollResponder.componentWillMount()`, same for `componentWillUnmount`, and then replace these lines: https://github.com/facebook/react-native/blob/master/Libraries/Components/ScrollView/ScrollView.js#L915-L936

with: 

```
const props = {
  // ... other props
  ...this._scrollResponder.getHandlers(),
  // ... other props
}
```

This should let as much as possible stay exactly the same. In fact changing the ScrollResponder in that way could probably be done without even switching off of createReactClass in the same PR. 👍

Reply:
If it's gonna be this little change, I could squeeze it into this PR, what do you think?

Reply:
We've had other things break when making changes to this kind of stuff. I'd really rather doing small PRs for this so that things can be bisected and reverted easily without necessarily having to revert the whole thing.

Reply:
So, this makes sense?:
1) get rid of mixin in ScrollView but keep the createReactClass 
2) rebase this and change to class component

Reply:
It potentially makes sense to do one between that just converts it to a class with the existing typings. We’ve run into problems with how we’ve handled migrating the native refs at that point which has caused lots of headaches. Having it be separate would probably make it easier to carefully test and review.

Reply:
Sorry for what seems like a lot of overhead here. We just got burned recently and I want to help make sure we don’t repeat the same issues and that you get to learn from our problems. :-)

Reply:
Oh yea, I'm totally ok with that. It caused me headaches when I was grabbing components by name/display name in tests (bad practice to do so tho) or snapshotting them :D